prov/tcp/src/tcpx.h

```diff
@@ -141,11 +141,18 @@ struct tcpx_rx_detect {
 
 typedef void (*tcpx_ep_progress_func_t)(struct tcpx_ep *ep);
 
+typedef int (*tcpx_rx_process_fn_t)(struct tcpx_xfer_entry *rx_entry);
+
+struct tcpx_cur_rx {
+	struct tcpx_xfer_entry	*rx_entry;
+	tcpx_rx_process_fn_t 	process_fn;
+};
```

Comment:
Is there a reason to create a sub-structure, versus embedding the 2 fields directly